infrastructure/adot.tf

```diff
@@ -0,0 +1,192 @@
+}
+
+# ---- Security group for Interface VPC Endpoints (tight inbound on 443) ----
+resource "aws_security_group" "telemetry_endpoints_sg" {
+  count       = var.create_vpc_endpoints ? 1 : 0
+  name        = "crossfeed-${var.stage}-telemetry-endpoints"
+  description = "Restrict access to X-Ray and CloudWatch Logs interface endpoints"
+  vpc_id      = local.vpc_id
+
+  ingress {
+    description = "HTTPS from within VPC"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = [local.vpc_cidr]
+  }
+
+  # Allow return traffic; scope to VPC CIDR
+  egress {
+    description = "Responses to VPC"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = [local.vpc_cidr]
+  }
+
+  tags = {
+    Project = var.project
+    Stage   = var.stage
+    Owner   = "Crossfeed managed resource"
+  }
+}
+
+# ---- Endpoint policies (limit use to our account) ----
+locals {
+  xray_vpce_policy = {
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Sid       = "AllowXRayWritesFromThisAccount",
+        Effect    = "Allow",
+        Principal = { AWS = local.account_root_arn },
+        Action = [
+          "xray:PutTraceSegments",
+          "xray:PutTelemetryRecords",
+          "xray:GetSamplingRules",
+          "xray:GetSamplingTargets",
+          "xray:GetSamplingStatisticSummaries"
+        ],
+        Resource = "*"
+      }
+    ]
+  }
+
+  logs_vpce_policy = {
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Sid       = "AllowCWLogsFromThisAccount",
+        Effect    = "Allow",
+        Principal = { AWS = local.account_root_arn },
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents",
+          "logs:DescribeLogGroups",
+          "logs:DescribeLogStreams"
+        ],
+        Resource = "*"
+      }
+    ]
+  }
+}
+
+# ---- Interface VPC Endpoints (toggle via create_vpc_endpoints) ----
+# X-Ray endpoint service name: com.amazonaws.${region}.xray
+resource "aws_vpc_endpoint" "xray" {
+  count               = var.create_vpc_endpoints ? 1 : 0
+  vpc_id              = local.vpc_id
+  service_name        = "com.amazonaws.${var.aws_region}.xray"
+  vpc_endpoint_type   = "Interface"
+  private_dns_enabled = true
+  subnet_ids          = local.subnets_ep
+  security_group_ids  = [aws_security_group.telemetry_endpoints_sg[0].id]
+  policy              = jsonencode(local.xray_vpce_policy)
+
+  tags = {
+    Project = var.project
+    Stage   = var.stage
+    Owner   = "Crossfeed managed resource"
+  }
+}
+
+# CloudWatch Logs endpoint service name: com.amazonaws.${region}.logs
+resource "aws_vpc_endpoint" "logs" {
+  count               = var.create_vpc_endpoints ? 1 : 0
+  vpc_id              = local.vpc_id
+  service_name        = "com.amazonaws.${var.aws_region}.logs"
+  vpc_endpoint_type   = "Interface"
+  private_dns_enabled = true
+  subnet_ids          = local.subnets_ep
```

Comment:
See database.tf we've been able to do this like this:
`subnet_ids = var.is_dmz ? [aws_subnet.db_1[0].id, aws_subnet.db_2[0].id] : [data.aws_ssm_parameter.subnet_db_1_id[0].value, data.aws_ssm_parameter.subnet_db_2_id[0].value]`

But would need to change to the groups you want